### ISSUE TITLE
batchGenerateCI() cannot handle tibbles

### ISSUE THREAD

**Issue by linjansen** [rdotsch/rcicr#70]
It appears that batchGenerateCI() cannot handle tibbles. It gave the following error message:

"Error in aggregate.data.frame(responses, by = list(stimuli = stimuli),  : 
 arguments must have same length"

When I saved the tibble as a dataframe (using as.data.frame) the problem was solved.

**Comment by aucuparia** [comment]
I hope I'm not being to blunt, but why should we support tibbles? Not that I have something against tibbles (in fact, I haven't used that format yet), but I don't know if it violates the "do one thing and do it well" software design principle. I'd gladly implement tibble support, but I want to make sure that we actually want to support it in rcicr itself.

@rdotsch, what's your take on this?

**Comment by rdotsch** [comment]
I agree but in order to make our package fool proof, we can just add a data.frame(tibble).

**Comment by aucuparia** [comment]
`batchGenerateCI` is going to be deprecated. See https://github.com/rdotsch/rcicr/issues/87. Once the refactor (#87) has been done, we can add tibble support to `generateCI` to resolve this issue.

**Comment by rdotsch** [comment]
This ended up popping up 6 years later 😄, see #123 .

**Comment by rdotsch** [comment]
**`generateCI()` cannot handle tibbles.**

Root cause was an `aggregate()` call that received a one-column tibble rather than a vector. `generateCI()`, `batchGenerateCI()` and `computeCumulativeCICorrelation()` now accept tibble columns as well as `data.frame` columns, and `generateCI()` additionally checks up front that `stimuli` and `responses` are the same length, so a mismatch reports itself clearly instead of surfacing as a confusing `aggregate` error.

This is fixed in `main`.

The package has just been through a substantial modernization pass: all seven known correctness bugs are fixed, there is now a test suite and CI, and every fix is covered by a regression test. A golden-master test pins the numeric output of the default pipeline to what it produced *before* the fixes, so default-configuration results are unchanged — that claim is backed by a test you can run, not by assertion.

```r
install.packages("remotes")
remotes::install_github("rdotsch/rcicr")
```

`install.packages("rcicr")` still does not work: the package was archived from CRAN in 2021 because the maintainer email address had gone stale. That address is fixed and a return to CRAN is being prepared, but GitHub is the source for now.

If you have results computed with an older version, please read the **"Reproducibility impact"** section of [`NEWS.md`](https://github.com/rdotsch/rcicr/blob/main/NEWS.md) before re-running — most fixes turn a crash into working behaviour and cannot change a number you already have, but two of them genuinely do change infoVal, and the conditions are spelled out there.

Thanks for reporting it, and apologies it sat for so long.
